x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/PowTests.java

```diff
@@ -39,71 +38,291 @@ public static Iterable<Object[]> parameters() {
+                    equalTo(0)
+                )
+            ),
+            new TestCaseSupplier(
+                "pow(123L, -1)",
+                () -> new TestCase(
+                    List.of(new TypedData(123L, DataTypes.LONG, "base"), new TypedData(-1, DataTypes.INTEGER, "exp")),
+                    "PowLongEvaluator[base=CastLongToDoubleEvaluator[v=Attribute[channel=0]], "
+                        + "exponent=CastIntToDoubleEvaluator[v=Attribute[channel=1]]]",
+                    DataTypes.LONG,
+                    equalTo(0L)
+                )
+            ),
+            new TestCaseSupplier(
+                "pow(123D, -1)",
+                () -> new TestCase(
+                    List.of(new TypedData(123.0, DataTypes.DOUBLE, "base"), new TypedData(-1, DataTypes.INTEGER, "exp")),
+                    "PowDoubleEvaluator[base=Attribute[channel=0], exponent=CastIntToDoubleEvaluator[v=Attribute[channel=1]]]",
+                    DataTypes.DOUBLE,
+                    equalTo(1D / 123D)
+                )
+            ),
+            new TestCaseSupplier("pow(integer, 1)", () -> {
+                int base = randomInt();
+                return new TestCase(
+                    List.of(new TypedData(base, DataTypes.INTEGER, "base"), new TypedData(1, DataTypes.INTEGER, "exp")),
+                    "PowIntEvaluator[base=CastIntToDoubleEvaluator[v=Attribute[channel=0]], "
+                        + "exponent=CastIntToDoubleEvaluator[v=Attribute[channel=1]]]",
+                    DataTypes.INTEGER,
+                    equalTo(base)
+                );
+            }),
+            new TestCaseSupplier(
+                "pow(1L, 1)",
+                () -> new TestCase(
+                    List.of(new TypedData(1L, DataTypes.LONG, "base"), new TypedData(1, DataTypes.INTEGER, "exp")),
+                    "PowLongEvaluator[base=CastLongToDoubleEvaluator[v=Attribute[channel=0]], "
+                        + "exponent=CastIntToDoubleEvaluator[v=Attribute[channel=1]]]",
+                    DataTypes.LONG,
+                    equalTo(1L)
+                )
+            ),
+            new TestCaseSupplier("pow(long, 1)", () -> {
+                // Avoid double precision loss
+                long base = randomLongBetween(-1L << 51, 1L << 51);
+                return new TestCase(
+                    List.of(new TypedData(base, DataTypes.LONG, "base"), new TypedData(1, DataTypes.INTEGER, "exp")),
+                    "PowLongEvaluator[base=CastLongToDoubleEvaluator[v=Attribute[channel=0]], "
+                        + "exponent=CastIntToDoubleEvaluator[v=Attribute[channel=1]]]",
+                    DataTypes.LONG,
+                    equalTo(base)
+                );
+            }),
+            new TestCaseSupplier("long-double overflow", () -> {
+                long base = 4339622345450989181L; // Not exactly representable as a double
+                long expected = 4339622345450989056L;
```

Comment:
This is a pretty concerning behavior.  For longs > 2^52, which are not exactly representable as doubles, x^1 != x.  I'm not sure what we should do about this.

Honestly, I think there's a strong argument that `pow` should always return a double (which is what Java does), since most powers of large longs will overflow anyway.